.github/workflows/self-hosted.yml

```diff
@@ -64,29 +73,148 @@ jobs:
 
-      - run: |
-          if [ ${{ github.event_name }} = 'push' ]; then
-            args=(
-              --tag ghcr.io/getsentry/sentry-self-hosted:latest
-              --push
-            )
+      - name: Download amd64 Image
+        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # v4.3.2
+        with:
+          name: sentry-amd64
+          path: /tmp
+
+      - name: Load amd64 Image
+        run: docker load --input /tmp/sentry-amd64.tar
+
+      - name: Download arm64 Image
+        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # v4.3.2
+        with:
+          name: sentry-arm64
+          path: /tmp
+
+      - name: Load arm64 Image
+        run: docker load --input /tmp/sentry-arm64.tar
+
+      - name: Push to GitHub Container Registry
+        run: |
+          docker tag sentry:amd64 ghcr.io/getsentry/sentry:${{ github.sha }}-amd64
+          docker push ghcr.io/getsentry/sentry:${{ github.sha }}-amd64
+
+          docker tag sentry:arm64 ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+          docker push ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+
+          docker manifest create \
+            ghcr.io/getsentry/sentry:${{ github.sha }} \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-amd64 \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+
+          docker manifest push ghcr.io/getsentry/sentry:${{ github.sha }}
+
+          docker manifest create \
+            ghcr.io/getsentry/sentry:latest \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-amd64 \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+
+          docker manifest push ghcr.io/getsentry/sentry:latest
+
+          docker manifest create \
+            ghcr.io/getsentry/sentry:nightly \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-amd64 \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+
+          docker manifest push ghcr.io/getsentry/sentry:nightly
+
+          # These manifests below are being used for development purposes.
+          # We shall keep it as is.
+          docker manifest create \
+            ghcr.io/getsentry/sentry-self-hosted:${{ github.sha }} \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-amd64 \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+
+          docker manifest push ghcr.io/getsentry/sentry-self-hosted:${{ github.sha }}
+
+          docker manifest create \
+            ghcr.io/getsentry/sentry-self-hosted:latest \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-amd64 \
+            --amend ghcr.io/getsentry/sentry:${{ github.sha }}-arm64
+
+          docker manifest push ghcr.io/getsentry/sentry-self-hosted:latest
+
+  publish-dockerhub:
+    needs: [assemble]
+    runs-on: ubuntu-latest
+    if: ${{ (github.ref_name == 'master') }}
```

Comment:
No `pull_request` limitation?

Reply:
Then why this check at all?

Reply:
Double guard :)

Reply:
Err, no please. Either have it full or none. No in between stuff.